### PR TITLE
lint: python: comparison with None

### PR DESCRIPTION
**Pull request by nrainer-materialize**
I first thought `== None` does not work (which is not the case) but I think the check still does not hurt.

### CODE REVIEW

File: ci/test/lint-main/checks/check-python-discouraged.sh [hunk 1]
```diff
@@ -18,28 +18,39 @@ cd "$(dirname "$0")/../../../.."
...
+      if [ -n "$TD_MULTI_LINE_MATCHES" ]; then
+          echo "$TD_MULTI_LINE_MATCHES"
       fi
 
-      exit 1
+      EXIT_CODE=1
+  fi
+
+  # check whether occurrences of `== None` exist
+  EQ_NONE_MATCHES=$(find misc/python/materialize test -name '*.py' -print0 | xargs -0 awk '/== None/ {print FILENAME ":" FNR}')
```
Comment (nrainer-materialize) [inline ci/test/lint-main/checks/check-python-discouraged.sh:46]: Using `awk` here to get the same style for free (`grep` would also work for this pattern)

**Review by def- — COMMENTED** [review]
Could we use one of our existing linters for this? For example https://www.flake8rules.com/rules/E711.html

**Comment by nrainer-materialize** [comment]
> Could we use one of our existing linters for this? For example https://www.flake8rules.com/rules/E711.html

Absolutely!

**Comment by nrainer-materialize** [comment]
Using the linter for this now, but still keeping the "refactoring" in ed64c4cfc0133a6a0a03d533448af7ac1135c4fb.

**Review by def- — APPROVED** [review]

**Pull request merged**